k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -172,12 +172,14 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
 
         case Some((event, ws)) =>
           import Ordering.Implicits._
-          // Register the update only if its resource version is larger than the largest version
+          // Register the update only if its resource version is larger than or equal to the largest version
           // seen so far.
-          if (largestEvent.forall(_ < event)) {
+          if (largestEvent.forall(_ <= event)) {
+            log.debug(s"k8s returned a new resource: $event")
             state.update(Activity.Ok(event))
             _processEventStream(ws, event.resourceVersion, Some(event))
           } else {
+            log.debug(s"k8s returned an old resource. Using most up to date resource: $event")
```

Comment:
TIOLI, but I think it would be nice if this log line included the following information:
+ the path of the watched resource
+ the received resource version
+ the previous resource version

If we log all this information, it might not be necessary to log the entire event as well (which can be somewhat wordy). Consider something like:

```scala
log.debug(
  "k8s watch on '%s' skipping event with resource version %s (older than most recent resource version %s)",
  watchPath,
  event.resourceVersion,
  largestEvent.map(_.resourceVersion)
)
```
or similar.

Reply:
Note also that the `sprintf`-style log formatting (`%s` and friends) is preferable over Scala string interpolation (`$name` etc) in twitter.util.logging, as it's only evaluated if that log level is enabled, which will have a reduced performance impact. See: https://github.com/twitter/util#using